shopfloor/services/cluster_picking.py

```diff
@@ -1122,17 +1122,19 @@ def set_destination_all(self, picking_batch_id, barcode, confirmation=False):
     def _unload_write_destination_on_lines(self, lines, location):
         lines.write({"shopfloor_unloaded": True, "location_dest_id": location.id})
         lines.package_level_id.location_dest_id = location
-        for line in lines:
-            # We set the picking to done only when the last line is
-            # unloaded to avoid backorders.
-            picking = line.picking_id
-            if picking.state == "done":
-                continue
-            picking_lines = picking.mapped("move_line_ids")
-            if all(line.shopfloor_unloaded for line in picking_lines):
-                picking._action_done()
+        for picking in lines.batch_id.picking_ids:
```

Comment:
@lmignon ,I replaced the loop on the lines with a loop on the pickings. I don't understand how it iterates multiple times on the lines